clients/mtaclient/retryable_mta_rest_client.go

```diff
@@ -153,7 +153,7 @@ func (c RetryableMtaRestClient) GetAsyncUploadJob(jobId string, namespace *strin
 	}
 	resp, err := baseclient.CallWithRetry(getAsyncUploadJobCb, c.MaxRetriesCount, c.RetryInterval)
 	if err != nil {
-		return AsyncUploadJobResult{}, err
+		return resp.(AsyncUploadJobResult), err
```

Comment:
So we can re-use the retry logic:
`
		if resp.StatusCode == 400 {
			return AsyncUploadJobResult{
				ClientActions: []string{"RETRY_UPLOAD"},
			}, fmt.Errorf("%s %s: %s, body: %s", couldNotGetAsyncJobError, jobId, resp.Status, string(bodyBytes))
		}
`
In this code the retry is explicitly set and it will be wiped out if we just return an empty object in the retryable client